src/snowcli/cli/snowpark/function/commands.py

```diff
@@ -201,25 +150,22 @@ def function_update(
             log.info(
                 "Return type or handler types do not match. Replacing the function."
             )
-            replace = True
+            create_or_replace = True
 
-    artifact_file = upload_snowpark_artifact(
+    artifact_file = _upload_snowpark_artifact(
         function_manager=fm,
-        stage_manager=sm,
         function_identifier=identifier,
         file=file,
-        overwrite=True,
     )
 
-    if replace:
+    if create_or_replace:
```

Comment:
I think we need to think about a bit longer.

1. Function does not exists, we upload artifacts and create the function
2. Function exists and has same signature - we upload only the artefact
3. Function exists but has different signature - we create a new function (so we and up with having two functions)

Reply:
I left logic as it was before. 

Before:
* create was putting the package and was creating a function/procedure
* update was always putting package and only if there was difference in packages or there was a change with handler or return type

After:
* deploy works as update